detector.go

```diff
@@ -298,12 +295,14 @@ func (bp *BuildpackTOML) Detect(c *DetectConfig) DetectRun {
 	}
 	if api.MustParse(bp.API).Equal(api.MustParse("0.2")) {
 		if t.hasInconsistentVersions() || t.Or.hasInconsistentVersions() {
-			t.Err = errInconsistentVersion
+			t.Err = errors.Errorf(errInconsistentVersion, bp.Buildpack.Name)
+			t.Code = CodeVersionFail
```

Comment:
I think we should using `t.Code = -1` here. We need to use a code that it would be impossible for the buildpack itself to return (negative error codes aren't valid).